### PR TITLE
test: Add testcases for Text Blocks introduced in Java 15

### PR DESCRIPTION
**Pull request by LakshyAAAgrawal**
Testcases from examples at https://openjdk.java.net/jeps/378
With this, the only other language changes for Java 15 will be preview features and hence Spoon can fully support Java 15.
Related  #3605

### CODE REVIEW

File: src/test/java/spoon/test/textBlocks/TextBlockTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class TextBlockTest{
+	private Launcher setUpTest() {
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/resources/textBlock/TextBlockTestClass.java");
+		launcher.run();
+		return launcher;
+	}
+	
+	@Test
+	public void testTextBlock(){
```
Comment (MartinWitt) [inline src/test/java/spoon/test/textBlocks/TextBlockTest.java:24]: split it in 3 testcases?

File: src/test/java/spoon/test/textBlocks/TextBlockTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		Launcher launcher = setUpTest();
+		CtClass<?> allstmt = (CtClass<?>) launcher.getFactory().Type().get("textBlock.TextBlockTestClass");
+		CtMethod<?> m1 =  allstmt.getMethod("m1");
+		CtStatement stmt1 = m1.getBody().getStatement(0);
+		CtLiteral l1 = (CtLiteral) stmt1.getDirectChildren().get(1);
```
Comment (MartinWitt) [inline src/test/java/spoon/test/textBlocks/TextBlockTest.java:29]: maybe use TypeFilter for more stable. Get(Number) is pretty unstable for a testcase. If we change order in any future case

File: src/test/java/spoon/test/textBlocks/TextBlockTest.java (hunk 1 above)
Comment (MartinWitt) [inline src/test/java/spoon/test/textBlocks/TextBlockTest.java:24]: In Spoon every testcase has a comment `//contract: ...` describing the testcase in natural language

**Comment by LakshyAAAgrawal** [comment]
Thanks a lot @MartinWitt 
Will make the necessary changes.

**Comment by MartinWitt** [comment]
Maybe add the empty textblock and the equal checks from jep?
Maybe add a bit unicode to the textblock?
Otherwise lgtm!

**Comment by MartinWitt** [comment]
A fix for the failing testcase could be pr #3630

**Comment by monperrus** [comment]
#3630 merged, we can rebase here

**Comment by LakshyAAAgrawal** [comment]
Will do immediately

**Comment by monperrus** [comment]
Thanks a lot!

Does this mean that text blocks are fully supported now?

Is there something else to do to say that Spoon is Java 15 compatible?

**Comment by LakshyAAAgrawal** [comment]
I just looked at the JDK 15 Release notes, and the only other language change I could think of testing was Unicode 13. I have added a symbol introduced in Unicode 13 in the test file. I think this completes the Java 15 compatibility, except for preview features.

**Comment by MartinWitt** [comment]
Have you tried printing a textblock and is it a textblock again? During my short tests i never tried this..

**Comment by LakshyAAAgrawal** [comment]
Yes.

On printing the method m1 from the resource test file, the following is the output:
```
void m1() {
    java.lang.String html = "<html>\n    <body>\n        <p>Hello, कसौटी 🥲</p>\n    </body>\n</html>\n";
}
```

**Comment by MartinWitt** [comment]
We should print a textblock if the input was a textblock. But I'm unsure how hard changing the printer is. If you want you can give it a try.

**Comment by monperrus** [comment]
> But merging it now and changing this detail later is fine for me.

Agree! Will merge this first.

**Pull request merged**